sql/src/main/java/org/apache/druid/sql/calcite/planner/PlannerFactory.java

```diff
@@ -191,7 +193,15 @@ public SqlConformance conformance()
               return null;
             }
           }
-        })
-        .build();
+        });
+
+    if (plannerConfig().getNativeQuerySqlPlanningMode()
+                       .equals(PlannerConfig.NATIVE_QUERY_SQL_PLANNING_MODE_DECOUPLED)
```

Comment:
`getNativeQuerySqlPlanningMode()` could maybe be null and generate an NPE.  There is perhaps a `@NotNull` annotation to try to protect from it, but you can just write your equals better and not have to rely on such things:

```
PlannerConfig.NATIVE_QUERY_SQL_PLANNING_MODE_DECOUPLED.equals(plannerConfig().getNativeSqlPlanningMode())
```

Reply:
thanks for the catch, fixed it!